samples/java-spring-eventsourced-customer-registry/src/test/java/customer/domain/CustomerEventSerializationTest.java

```diff
@@ -0,0 +1,95 @@
+package customer.domain;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import customer.domain.schemaevolution.AddressChangedMigration;
+import customer.domain.schemaevolution.NameChangedMigration;
+import kalix.javasdk.JsonSupport;
+import org.junit.jupiter.api.Test;
+
+import java.util.Base64;
+import java.util.Optional;
+
+import static customer.domain.schemaevolution.CustomerEvent.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+class CustomerEventSerializationTest {
+
+  @Test
+  public void shouldDeserializeWithOptionalField() {
+    //given
+    Any serialized = JsonSupport.encodeJson(new CustomerEvent.NameChanged("andre"));
+
+    //when
+    NameChanged deserialized = JsonSupport.decodeJson(NameChanged.class, serialized);
+
+    //then
+    assertEquals("andre", deserialized.newName());
+    assertEquals(Optional.empty(), deserialized.oldName());
+    assertEquals(null, deserialized.reason());
+  }
+
+  @Test
+  public void shouldDeserializeWithMandatoryField() {
+    //given
+    Any serialized = JsonSupport.encodeJson(new CustomerEvent.NameChanged("andre"));
+
+    //when
+    NameChanged deserialized = JsonSupport.decodeJson(NameChanged.class, serialized, Optional.of(new NameChangedMigration()));
+
+    //then
+    assertEquals("andre", deserialized.newName());
+    assertEquals(Optional.empty(), deserialized.oldName());
+    assertEquals("default reason", deserialized.reason());
+  }
+
+  @Test
+  public void shouldDeserializeWithChangedFieldName() {
+    //given
+    Address address = new Address("Wall Street", "New York");
+    Any serialized = JsonSupport.encodeJson(new CustomerEvent.AddressChanged(address));
+
+    //when
+    AddressChanged deserialized = JsonSupport.decodeJson(AddressChanged.class, serialized, Optional.of(new AddressChangedMigration()));
+
+    //then
+    assertEquals(address, deserialized.newAddress());
+  }
+
+  @Test
+  public void shouldDeserializeWithStructureMigration() {
+    //given
+    Any serialized = JsonSupport.encodeJson(new CustomerCreated("bob@lightbend.com", "bob", "Wall Street", "New York"));
+
+    //when
+    CustomerEvent.CustomerCreated deserialized = JsonSupport.decodeJson(CustomerEvent.CustomerCreated.class, serialized, Optional.of(new CustomerCreatedMigration()));
+
+    //then
+    assertEquals("Wall Street", deserialized.address().street());
+    assertEquals("New York", deserialized.address().city());
+  }
+
+  // tag::testing-deserialization[]
+  @Test
+  public void shouldDeserializeCustomerCreated_V0() throws InvalidProtocolBufferException {
+    // end::testing-deserialization[]
+    Any serialized = JsonSupport.encodeJson(new CustomerCreated("bob@lightbend.com", "bob", "Wall Street", "New York"));
+
+    new String(Base64.getEncoder().encode(serialized.toByteArray()));
```

Comment:
This include is working, but strange enough the lines be included are lines 77, 78 and 79 of this file. Something is broken here.